relays/substrate/src/rialto_millau/millau_messages_to_rialto.rs

```diff
@@ -131,11 +131,30 @@ pub async fn run(
 	rialto_client: RialtoClient,
 	rialto_sign: RialtoSigningParams,
 	lane_id: LaneId,
+	enable_headers_relay: bool,
 	metrics_params: Option<MetricsParams>,
 ) {
 	let stall_timeout = Duration::from_secs(5 * 60);
 	let relayer_id_at_millau = millau_sign.signer.public().as_array_ref().clone().into();
 
+	let rialto_to_millau_headers_relay = if enable_headers_relay {
+		Some(crate::rialto_millau::rialto_headers_to_millau::on_demand(
+			rialto_client.clone(),
+			millau_client.clone(),
+			millau_sign.clone(),
+		))
+	} else {
+		None
+	};
```

Comment:
Sharing since I just learned about this yesterday: [`bool::then`](https://doc.rust-lang.org/std/primitive.bool.html#method.then) was introduced in Rust `v1.50.0`, so this could be rewritten as:

```rust
 let rialto_to_millau_headers_relay = enable_headers_relay.then(|| {
     crate::rialto_millau::rialto_headers_to_millau::on_demand(
         rialto_client.clone(),
         millau_client.clone(),
         millau_sign.clone(),
     )
 });
```

I'm still unsure as to how I feel about it though, since it's quite "functional"